File: number_theory/integer_factorization.cpp
// integer_factorization.cpp - Naive prime factorization.
// Author: Scott

#include <iostream>
using namespace std;

void factorize(int n) {
    cout << "Prime factors of " << n << ": ";
    for (int i = 2; i*i <= n; ++i) {
        while (n % i == 0) {
            cout << i << " ";
            n /= i;
        }
    }
    if (n > 1) cout << n;
    cout << endl;
}

int main() {
    int n = 100;
    cout << "Enter number to factor (default 100): ";
    string input;
    getline(cin, input);
    if (!input.empty()) n = stoi(input);

    factorize(n);
    return 0;
}

File: number_theory/integer_factorization.py
"""
integer_factorization.py - Naive trial division to factor an integer into primes.
Good educational intro to factoring.
Author: Scott
"""

def factorize(n):
    i = 2
    factors = []
    while i * i <= n:
        while n % i == 0:
            factors.append(i)
            n //= i
        i += 1
    if n > 1:
        factors.append(n)
    return factors

if __name__ == "__main__":
    n = int(input("Enter number to factor: ") or "84")
    print(f"Prime factors of {n}: {factorize(n)}")
